src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrFromCopyState = [];
+  const copyState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(copyState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const prop of action.keysToRemove) {
```

Comment:
```suggestion
        for (const key of action.keysToRemove) {
```
or `key`, or `keyToRemove` would be better (prop is a pair of key and value)